### PR TITLE
perform 'full' session restart after activate

### PR DESCRIPTION
**Pull request by kevinushey**
Closes https://github.com/rstudio/renv/issues/1530.

### CODE REVIEW

File: R/restart.R [hunk 1]
```diff
@@ -49,12 +49,17 @@ renv_restart_request_rstudio <- function(project, reason, ...) {
...
+  # the 'restart' option here as the version RStudio uses
+  # tries to preserve session state that we need to change.
+  #
+  # https://github.com/rstudio/renv/issues/1530
   projdir <- tools$.rs.getProjectDirectory() %||% ""
   if (renv_file_same(projdir, project)) {
-    restart <- getOption("restart")
-    if (is.function(restart))
-      return(restart())
+    restart <- getOption("renv.restart.function", default = function() {
```
Comment (hadley) [inline R/restart.R:59]: Is this a new option?

Comment (kevinushey) [reply R/restart.R:59]: Yeah. It's really just an escape hatch in case someone needs it for some reason.

File: R/restart.R [hunk 2]
```diff
@@ -49,12 +49,17 @@ renv_restart_request_rstudio <- function(project, reason, ...) {
...
+      .rs.api.executeCommand("restartR", quiet = TRUE)
+    })
+    return(restart())
```
Comment (hadley) [inline R/restart.R:62]: I wonder if this function should return invisibly because it's called primarily for its side effects.

**Review by hadley — APPROVED** [review]

File: R/restart.R (hunk 2 above)
Comment (kevinushey) [reply R/restart.R:62]: It should; thanks for catching that!

**Pull request merged**